src/kbmod_wf/task_impls/kbmod_search.py

```diff
@@ -14,3 +19,70 @@ def kbmod_search(input_wu=None, result_file=None, logger=None):
         f.write(f"Logged: {value} - {time.time()}\n")
 
     return result_file
+
+
+def kbmod_search(
+    wu_filepath: str = None,
+    search_config_filepath: str = None,  # ! Seems like this should be provided in the runtime_config dictionary ???
+    result_filepath: str = None,
+    runtime_config: dict = {},
+    logger: Logger = None,
+):
+    kbmod_searcher = KBMODSearcher(
+        wu_filepath=wu_filepath,
+        search_config_filepath=search_config_filepath,
+        result_filepath=result_filepath,
+        runtime_config=runtime_config,
+        logger=logger,
+    )
+
+    return kbmod_searcher.run_search()
+
+
+class KBMODSearcher:
+    def __init__(
+        self,
+        wu_filepath: str = None,
+        search_config_filepath: str = None,
+        result_filepath: str = None,
+        runtime_config: dict = {},
+        logger: Logger = None,
+    ):
+        self.input_wu_filepath = wu_filepath
+        self.search_config_filepath = search_config_filepath
+        self.runtime_config = runtime_config
+        self.output_filepath = result_filepath
+        self.logger = logger
+
+    def run_search(self):
+        self.logger.info("Loading workunit from file")
+        wu = WorkUnit.from_fits(self.input_wu_filepath)
+
+        self.logger.debug("Loaded work unit")
+        if self.search_config is not None:
+            # Load a search configuration, otherwise use the one loaded with the work unit
+            wu.config = kbmod.configuration.SearchConfiguration.from_file(self.search_config)
+
+        config = wu.config
+
+        # Modify the work unit results to be what is specified in command line args
+        input_parameters = {
+            "res_filepath": args.result_dir,
+            "result_filename": os.path.join(args.result_dir, "full_results.ecsv"),
+        }
+        config.set_multiple(input_parameters)
+
+        # Save the search config in the results directory for record keeping
+        config.to_file(os.path.join(args.result_dir, "search_config.yaml"))
+        wu.config = config
+
+        self.logger.info("Running KBMOD search")
+        res = kbmod.run_search.SearchRunner().run_search_from_work_unit(wu)
+
+        self.logger.info("Search complete")
+        self.logger.info(f"Search results: {res}")  # ! Should this be logged to a file ???
```

Comment:
@wilsonbb I'm not sure what `res` is here, but do we really need to log it? I'm concerned that this might be a large thing to include in the logs.

Reply:
This is a KBMOD [Results object](https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/results.py#L21) which wraps around an astropy Table. It might make sense to log `len(res)` but I agree that we shouldn't have to log the whole table

Reply:
Updated to len(res)